### PR TITLE
Feature/add missing fn trace

### PR DESCRIPTION
**Pull request by dizzzz**
Adding fn:trace#1 function
Testcases are included

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java [hunk 1]
```diff
@@ -38,40 +39,53 @@
  */
 public class FunTrace extends BasicFunction {
     
-    public final static FunctionSignature signature =
+    public final static FunctionSignature[] signatures = {
```
Comment (adamretter) [inline exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java:42]: Any interest in using the newer FunctionDSL approach? It has some advantages...

**Review by adamretter — APPROVED** [review]
Just one comment.

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java (hunk 1 above)
Comment (dizzzz) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java:42]: I was in doubt to apply that pattern, but was not sure how to start....

Comment (dizzzz) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java:42]: yes lets do that, do you have a good sample for it?

**Review by joewiz — APPROVED** [review]
@dizzzz Thank you for helping eXist move toward completing eXist's coverage of the XQuery 3.1 spec! I ran this PR against https://github.com/w3c/qt3tests/blob/master/fn/trace.xml, using https://github.com/eXist-db/exist-xqts-runner (commands: `sbt clean compile`, `sbt assembly`, and `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar -ts fn-trace`). With this PR, eXist passes 24 of the 27 tests for `fn:trace` (see [fn-trace-junit-data-PR-3597.zip](https://github.com/eXist-db/exist/files/5471319/fn-trace-junit-data-PR-3597.zip)). Surprisingly, develop passes 25 of the 27 tests (see [fn-trace-junit-data-develop.zip](https://github.com/eXist-db/exist/files/5471320/fn-trace-junit-data-develop.zip)). However, scanning through the failures, it seems to me that none of them appear to be consequential.

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java (hunk 1 above)
Comment (adamretter) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FunTrace.java:42]: I think the CacheModule isn't a bad example.

**Comment by dizzzz** [comment]
image

**Comment by dizzzz** [comment]
updated description @adamretter @joewiz

**Review by adamretter — APPROVED** [review]
@dizzzz Nicely done :-) How did you find the FunctionDSL?

**Comment by dizzzz** [comment]
It gives more structure than the original WoW I guess. I will start using it from now :-)

**Comment by adamretter** [comment]
@dizzzz What does "WoW" stand for?

I also just pushed a very small improvement ;-)

**Comment by dizzzz** [comment]
Way Of Working :-P

**Review by dizzzz — COMMENTED** [review]
LGTM

**Pull request merged**